hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/store/ProcedureStore.java

```diff
@@ -240,4 +240,10 @@ default void forceUpdate(long[] procIds) {
    * @param count the number of IDs to delete
    */
   void delete(long[] procIds, int offset, int count);
+
+  /**
+   * Will be called by the framework to give the store a chance to do some clean up works.
+   */
+  default void cleanup() {
```

Comment:
Ah this is not for close the store... The way to close the store is through the stop method above.

Reply:
ok. why we need the cleanup then? Can't do it in the close?

Reply:
I should add some comments on the cleanup method of RegionProcedureStore. In fact, I think a better design is to let the RegionProcedureStore to handle this by itself but adds a 'cleanup' call to the end of the existing cleanup job will cost less resources.

The story is, we need to keep the maxProcId to avoid the procedure d go backwards after master restarts, so when deleting a procedure, now we just fill the column with an empty array, and this cleanup is to cleanup the unused deleted procedure rows, as we only need to retain the row with max procedure id.

You can see the serializeDelete and cleanup method in RegionProcedureStore to better understand the logic here.

If you do not like this, another way is to schedule a background cleanup task in RegionProcedureStore, it will make the interface clean, but we need one more thread doing cleanup work then.

Reply:
Add comment on how cleanup works and let it at that. Lets not add a background thread. Thanks.